manic/repository_git.py

```diff
@@ -250,7 +268,7 @@ def _determine_remote_name(self):
             data = data.split()
             name = data[0].strip()
             url = data[1].strip()
-            if self._url == url:
+            if self._url == url or not self._port is EMPTY_STR:
```

Comment:
Is this logic really correct? It looks to me like, if self._port is not the empty string, then this will treat any url as a match. This seems wrong if there are multiple remotes: won't this just pick the first one all the time?

It's possible that the need for this change would go away with my above suggestions (the comment beginning, "It feels wrong to me to update self._url here").

Reply:
Also: I believe `XXX is not` is preferred to `not XXX is` (see https://www.python.org/dev/peps/pep-0008/#programming-recommendations)